src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4848,6 +4871,7 @@ internal Socket UpdateAcceptSocket(Socket socket, EndPoint remoteEP)
             socket._protocolType = _protocolType;
             socket._rightEndPoint = _rightEndPoint;
             socket._remoteEndPoint = remoteEP;
+            socket._localEndPoint = null;
```

Comment:
It seems like here we could get the local end point from the listener, right?

Reply:
If I do
```c#
socket._localEndPoint = _localEndPoint;
```
then some tests fail, for example in `DualModeBeginAccept.BeginAcceptV4BoundToAnyV4_Success` with address being `::ffff:0.0` or so instead of expected `::ffff:127.0.0.1` here https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs#L857
Or you were suggesting something else?

Reply:
Yeah, that's what I was suggesting.

How do the tests fail? Is it only dual mode tests that fail?

The code as-is seems fine, but if there's a reason we can't do my suggestion, it would probably be good to add a comment explaining why so that we don't try to "optimize" this in the future and hit the same issue.

Reply:
I've tried it now and didn't get these failures as I did before 😞 It seems that one of my later changes fixed it. So I added the assignment. Sorry for misinformation.

Reply:
So the test failures are actually back in CI. I will do some more investigation to understand the behavior of sockets there.

Reply:
This is really weird, looks like a wildcard address is being set to `_localEndpoint` in the listener socket, which is then copied to the accept socket. If you can get to the bottom of it, it would give us some good insight.

@geoffkizer it looks common sense that the address shall be copied to the accept socket, but ideally I would like to see `getsockname` and `accept` docs guaranteeing this, especially if there may be weird corner cases happening around dual mode.

Reply:
I'm fine with that. I mostly just want to make sure we understand this and comment it appropriately in the code. The wildcard binding case seems like good one to understand and comment about in the code.